### PR TITLE
Add docs, changelog and authors in CONTRIBUTION and fix typos and errors

### LINKED ISSUE #2252: Update contribution guidelines to include doc, changelogs and authors
This would help keeping these up to date!

### PR DESCRIPTION
**Pull request by shricodev**
Add docs, changelog, and authors in Development. Also fixed some minor typos and errors.

Hopefully, this fixes #2252 

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

Signed-off-by: Shrijal Acharya <coveringguy0123@gmail.com>

### CODE REVIEW

**Comment by shricodev** [comment]
How do I fix these errors?

**Comment by AyanSinhaMahapatra** [comment]
You don't need to worry about the failing test at ` nexB.scancode-toolkit (core_tests misc_and_scancode) ` as this is unrelated to your changes and will be fixed by https://github.com/nexB/scancode-toolkit/pull/3202/commits/d4fed8703b7e378784e9574c58c06d6166dc7715

**Comment by OctoPie23** [comment]
I was kinda worried.😂
Thanks a lot, @AyanSinhaMahapatra 😌
Is this getting merged any sooner?

**Comment by shricodev** [comment]
@AyanSinhaMahapatra Sir, gentle ping. Do you happen to have any updates on this one?

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -54,7 +52,7 @@ For other questions, discussions, and chats, we have:
   This is a busy place with a lot of CI and commit notifications that makes
   actual chat sometimes difficult!
 
-- a mailing list at `sourceforge <https://lists.sourceforge.net/lists/listinfo/aboutcode-discuss>`_
+- a mailing list at `SourceForge <https://lists.sourceforge.net/lists/listinfo/aboutcode-discuss>`_
```
Comment (AyanSinhaMahapatra) [inline CONTRIBUTING.rst]: I couldn't find anything at https://lists.sourceforge.net/lists/listinfo/aboutcode-discuss, this is old and maybe we want to remove this?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -14,17 +14,15 @@ check `easy issues <https://github.com/nexB/scancode-toolkit/labels/easy>`_
...
 For commits, it is best to simply add a line like this to your commit message,
 with your name and email::
 
     Signed-off-by: Jane Doe <developer@example.com>
 
 Please try to write a good commit message, see `good commit message wiki
-<https://aboutcode.readthedocs.io/en/latest/contributing/writing_good_commit_messages.html>` for
-details. In particular use the imperative for your commit subject: think that
-you are giving an order to the codebase to update itself.
+<https://aboutcode.readthedocs.io/en/latest/contributing/writing_good_commit_messages.html>`_ for details. In particular use the imperative for your commit subject: think that you are giving an order to the codebase to update itself.
```
Comment (AyanSinhaMahapatra) [inline CONTRIBUTING.rst]: why put this on a single line instead of three?

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
Thanks, see comments

File: CONTRIBUTING.rst (hunk 1 above)
Comment (shricodev) [reply CONTRIBUTING.rst]: Is there an alternative, or should this be removed entirely?
 I've removed it for the time being, but if any new mailing lists appear, I'll update them.

File: CONTRIBUTING.rst (hunk 2 above)
Comment (shricodev) [reply CONTRIBUTING.rst]: Formatting issue. Fixed!

**Comment by AyanSinhaMahapatra** [comment]
@OctoPie23 can you rebase this into one commit and add a signoff so the DCO is passing properly?

**Comment by shricodev** [comment]
Sir, I have been away from home for three days. It was necessary. My laptop
isn't with me right now. I'll do it as soon as I get back.
Hope you understand :)

**Comment by shricodev** [comment]
> @OctoPie23 can you rebase this into one commit and add a signoff so the DCO is passing properly?

@AyanSinhaMahapatra Sure! 
Shall we consider merging this now?

**Review by AyanSinhaMahapatra — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**